web/src/components/datatable.tsx

```diff
@@ -0,0 +1,77 @@
+import { For, JSX, Show, createMemo, createSignal } from 'solid-js';
+
+export interface DataTableColumn<T> {
+  label: string;
+  get: (x: T) => JSX.Element;
+  sortBy?: (x: T) => string;
+}
+export interface DataTableProps<T> {
+  columns: DataTableColumn<T>[];
+  items: T[];
+}
+export function DataTable<T>(props: DataTableProps<T>): JSX.Element {
+  const [sortComparator, setSortComparator] = createSignal<
+    ((x: T, y: T) => number) | undefined
+  >();
+  const sortedItems = createMemo(() => {
+    const currentSortComparator = sortComparator();
+    if (currentSortComparator == undefined) {
+      return props.items;
+    } else {
+      const items = [...props.items];
+      items.sort(currentSortComparator);
+      return items;
+    }
+  });
```

Comment:
Add a newline between elements. This should help readability.

```suggestion
export interface DataTableColumn<T> {
  label: string;
  get: (x: T) => JSX.Element;
  sortBy?: (x: T) => string;
}

export interface DataTableProps<T> {
  columns: DataTableColumn<T>[];
  items: T[];
}

export function DataTable<T>(props: DataTableProps<T>): JSX.Element {
  const [sortComparator, setSortComparator] = createSignal<
    ((x: T, y: T) => number) | undefined
  >();
  
  const sortedItems = createMemo(() => {
    const currentSortComparator = sortComparator();
    if (currentSortComparator == undefined) {
      return props.items;
    } else {
      const items = [...props.items];
      items.sort(currentSortComparator);
      return items;
    }
  });
```